PAB-4250: Add info on Notifications 2.0 to the Streaming Analytics doc (#2409)
SoftwareAG/c8y-docs: content/change-logs/analytics/apama-in-c8y-20240902-Notifications-2.0.md

---
date:
title: Streaming Analytics and Notifications 2.0
change_type:
  - value: change-QHu1GdukP
    label: Feature
product_area: Analytics
component:
  - value: component-M5-cepIIS
    label: Streaming Analytics
build_artifact:
  - value: tc-KXXmo2SUR
    label: apama-in-c8y
ticket: PAB-4250
version:
---

> Streaming Analytics can now use the {{< product-c8y-iot >}} Notifications 2.0 reliable data forwarding capability to receive update notifications on new measurements, events, alarms, managed objects and operations that are processed by the {{< product-c8y-iot >}} platform. This feature is currently in private preview, so to take advantage of it, you must contact {{< product-c8y-iot >}} Operations to set the `notification2.streaming-analytics` feature flag. In addition, users of any of the variants of the Apama-ctrl microservice must also toggle the `streaming-analytics.messaging` feature flag. Users of custom microservices must also add the `ROLE_NOTIFICATION_2_ADMIN` permission to the microservice manifest. Users of external Apama deployments must add the new **Cumulocity Notifications 2.0** bundle to their Apama projects. For more information, including how to toggle the feature flags, see [Configuration requirements for Notifications 2.0](/streaming-analytics/analytics-customization/#notifications).

Reviewer: you must contact {{< product-c8y-iot >}} Operations

Shouldn´t we better say "product support" as we always do and link to the support page in the documentation https://cumulocity.com/docs/additional-resources/contacting-support/? Or how would customers contact "Operations"?

Author: Thanks Beate, I've changed this as you suggested.

> Streaming Analytics can now use the {{< product-c8y-iot >}} Notifications 2.0 reliable data forwarding capability to receive update notifications on new measurements, events, alarms, managed objects and operations that are processed by the {{< product-c8y-iot >}} platform. This feature is currently in private preview, so to take advantage of it, you must contact {{< product-c8y-iot >}} Operations to set the `notification2.streaming-analytics` feature flag. In addition, users of any of the variants of the Apama-ctrl microservice also need to toggle the `streaming-analytics.messaging` feature flag. All users of both Apama-ctrl and custom microservices must also add the `ROLE_NOTIFICATION_2_ADMIN` permission to the microservice manifest. Users of external Apama deployments must add the new **Cumulocity Notifications 2.0** bundle to their Apama projects. For more information, including how to toggle the feature flags, see [Configuration requirements for Notifications 2.0](/streaming-analytics/analytics-customization/#notifications).

Reviewer 2: We need a "why" statement here - and I need R&D to help with the details for this one. We should not have any change long entry without a "why" statement.

There are two distinct use cases - using Streaming Analytics within the platform and using it outside the platform. Outside the platform, the new capability replaces the CometD real-time notifications API which had robustness issues including sometimes not reconnecting after an outage. Why should a customer just using Streaming Analytics within the platform make the change to Notifications 2.0?

---
date:
title: Streaming Analytics and Notifications 2.0
change_type:
  - value: change-QHu1GdukP
    label: Feature
product_area: Analytics
component:
  - value: component-M5-cepIIS
    label: Streaming Analytics
build_artifact:
  - value: tc-KXXmo2SUR
    label: apama-in-c8y
ticket: PAB-4250
version:
---

> Streaming Analytics can now use the {{< product-c8y-iot >}} Notifications 2.0 reliable data forwarding capability to receive update notifications on new measurements, events, alarms, managed objects and operations that are processed by the {{< product-c8y-iot >}} platform.

Reviewer 3: ```suggestion
Streaming Analytics can now use the {{< product-c8y-iot >}} Notifications 2.0 reliable data forwarding capability to receive notifications for measurements, events, alarms, managed objects and operations that are processed by the {{< product-c8y-iot >}} platform.
```

> The benefits of using Notifications 2.0 are improved performance and reliability for messaging with Streaming Analytics applications such as smart rules, Analytics Builder and EPL apps. In particular, Apama EPL projects that remotely connect to Cumulocity IoT via the long-polling interface will benefit from improved robustness when migrating to Notifications 2.0.

Reviewer 2: "improved robustness when migrating"
Should be "after" not "when".

> The benefits of using Notifications 2.0 are improved performance and reliability for messaging with Streaming Analytics applications such as smart rules, Analytics Builder and EPL apps. In particular, Apama EPL projects that remotely connect to Cumulocity IoT via the long-polling interface will benefit from improved robustness after migrating to Notifications 2.0.

Reviewer 3: I think we need to split the personas here. Lets not mix the streaming analytics usecases with Apama EPL projects. Perhaps move the stuff after `In particular, Apama EPL projects ...` to later part of this entry

Author: Moved this. Better now?

> Streaming Analytics intends to migrate all tenants to use Notifications 2.0, and some customers may choose to start migrating now. This change is intended to provide improved performance, reliability and resilience in planned future enhancements to Streaming Analytics.

Reviewer 3: ```suggestion
Streaming Analytics intends to eventually migrate all tenants to use Notifications 2.0, and some customers may choose to start migrating now. This change is intended to provide improved performance, reliability and resilience in planned future enhancements to Streaming Analytics.
```

Author: Please see Rob's resolved comment above. where he said this: 
Should be "intends to" instead of "will eventually".
So we should not add this word again.

Reviewer 3: sorry, Ignore me

> Streaming Analytics will eventually migrate all tenants to use Notifications 2.0, and some customers may choose to start migrating now. This change is intended to provide improved performance, reliability and resilience in planned future enhancements to Streaming Analytics.

Reviewer 2: "Streaming Analytics will eventually"
Should be "intends to" instead of "will eventually".

> This feature is currently in private preview. To take advantage of it, you must contact [product support](/additional-resources/contacting-support/) to set the `notification2.streaming-analytics` feature flag. In addition, users of any of the variants of the Apama-ctrl microservice must also toggle the `streaming-analytics.messaging` feature flag. Users of custom microservices must also add the `ROLE_NOTIFICATION_2_ADMIN` permission to the microservice manifest. Users of external Apama deployments must add the new **Cumulocity Notifications 2.0** bundle to their Apama projects. For more information, including how to toggle the feature flags, see [Configuration requirements for Notifications 2.0](/streaming-analytics/analytics-customization/#notifications).

Reviewer 3: ```suggestion
This feature is currently in private preview. To take advantage of it, you must contact [product support](/additional-resources/contacting-support/) to set the `notification2.streaming-analytics` feature flag. In addition, users of any of the variants of the Apama-ctrl microservice must also toggle the `streaming-analytics.messaging` feature flag. Users of custom Apama microservices must also add the `ROLE_NOTIFICATION_2_ADMIN` permission to the microservice manifest. Users of custom Apama EPL projects must also add the new **Cumulocity Notifications 2.0** bundle. For more information, including how to toggle the feature flags, see [Configuration requirements for Notifications 2.0](/streaming-analytics/analytics-customization/#notifications).
```
